src/main/java/com/google/swarm/tokenization/common/Util.java

```diff
@@ -83,6 +84,11 @@ public enum FileType {
     TSV
   }
 
+  public enum InputLocation {
```

Comment:
I think introduction of this enum is making the code a little more complicated. From what I understood, if the input source is GCS, then we want to use pubsub mechanism instead of polling to discover new streamed files. And in case source is not GCS, we will continue using polling for new streamed files. I think we should try to answer if only processing new streamed files and disabling running pipeline on existing data a practical use case or not. In my opinion, we should always be processing the existing data in storage. Additionally, if required, new streamed files can be processed using pubsub if configured.

Please let me know if I am missing anything.

Reply:
Resolving as discussed offline